### DISCUSSION TITLE
I got some questions about Cookies Notices

### DISCUSSION THREAD

**Discussion by GitAcc53** [uBlockOrigin/uAssets#19262] [Q&A]
I use VPN and restart my PC a lot and use incognito sometimes so I need to get rid of these notices.

Cookies Notices doesnt work on most websites. It works on Google.com and youtube.com but it doesnt work on twitch and reddit and many other websites. So when I say "it worked" in my questions, it means it barely worked. Here are my questions :

1- After a fresh install on Opera, clicking Apply changes on "uBlock filters -- Cookies Notices" didnt work. But Update now worked. On Chrome, Edge and Firefox, Apply changes worked. On Opera, only Update now worked. But after another fresh install, Apply changes worked on Opera also. Why? 

2- Even after fresh install, everytime I tick Cookie Notices, "Update now" comes up. When I turn it off, does it delete itself? Is that why it updates itself everytime I tick it?

3- Ticking cookie notices "rarely may cause problems with scrolling or blanking pages/content" it says. How rarely?

4-uBlock filters Cookie Notices works, Easylist Cookie Notices doesnt work. Why? Apparently uBlock filters Cookie Notices was recently added with version 1.51.0 , which means I couldnt get rid of Cookie Notices until recently. Why doesnt Easylist Cookie Notices work?

5- Whats the difference between Apply Changes and Update now? When I click Apply Changes, Update now option goes away.

6- I didnt see "uBlock filters -- Cookies Notices" option before fresh install. Only "EasyList -- Cookies Notices". Why?

7- Someone on reddit said "Purge all caches is the most important, it's to mark all your filter lists as outdated all, and Update now will force update all of them. Purge all caches -> Update now is to make sure you have latest interface and up-to-date lists." but I didnt understand what he meant. Instead of messing around with "Purge all caches", can I just reinstall uBlock Origin to update it? Because I dont even know when should I push that "Purge all caches" button.

8- On some sites, I see a very very short moment, a flash of the page with cookie notice, than less then a second later, the page changes. Why?

edit : I just tried with both "Easylist -- Cookies Notices" and "uBlock filters -- Cookie Notices" options on, it worked I think.  
"uBlock filters -- Cookie Notices" removed google.com and youtube.com cookie notices and "Easylist -- Cookies Notices" removed the cookie notices of other websites. 

edit2: Its driving me crazy. On with both "Easylist -- Cookies Notices" and "uBlock filters -- Cookie Notices" enabled, with VPN in incognito, google.com and translate.google.com has cookie notices on both Opera and Chrome. Other site seems ok for now. The thing is its not consistent. I close the incognito tab and reopen it again, sometimes translat.google has notice, sometimes google.com has notice. I am thinking of just disabling both options.

**Comment by ghost** [comment]
https://www.reddit.com/r/uBlockOrigin/comments/15fambo/fastpicorg_has_so_many_ads/jui38bp/

**Reply by GitAcc53** [reply]
Are you him?

**Reply by GitAcc53** [reply]
I didnt understand some of the answers so I asked again

**Reply by stephenhawk8054** [reply]
@GitAcc53 No, this is my github account.

**Comment by stephenhawk8054** [comment]
> Because I dont even know where should I push that "Purge all caches" button.

@GitAcc53 Can you screenshot your "Filter lists" pane here?

**Reply by GitAcc53** [reply]
https://i.imgur.com/Bepf2ID.png

**Reply by stephenhawk8054** [reply]
It's right next to the "Update now" button 😐

**Reply by GitAcc53** [reply]
I meant when* sorry. I dont understand its function thats all.

**Reply by stephenhawk8054** [reply]
When you want to force update all your filter lists to latest versions.

**Reply by GitAcc53** [reply]
What is update for than? Also why isnt it latest version with fresh install? Why do we have to mess around with options?

**Reply by stephenhawk8054** [reply]
> What is update for than?

I don't understand

> why isnt it latest version with fresh install?

The filter lists are auto-updated few minutes after installing

> Why do we have to mess around with options?

When you have some issues, it could be that the issue is already fixed but your current filter lists have not updated to the fix yet because the filter lists could take some days for auto-updating.

And many users don't want to reinstall the whole extension just for force updating, because you will lose your current settings (for example, you have to tick on the cookie filter lists again after installing, right?).

---

Ok, I won't participate in here any more, sorry. Good luck.

**Reply by GitAcc53** [reply]
Ok I'll ask just in case someone else answers.

If purge updates filters, than what does update do?

"When you have some issues, it could be that the issue is already fixed but your current filter lists have not updated to the fix yet because the filter lists could take some days for auto-updating."

But I see "update" after clicking on filters even after fresh install. Even after 30 minutes after fresh install. Also my issues arent fixed at all.

**Reply by gothic-bum** [reply]
Per the [documentation](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists), which I strongly recommend that you fully read:
  
> **Purge all caches**
> This will reset the "last update" timestamp for all of the subscribed filter lists. Essentially, this will cause all filter lists to become out of date. This can be used to force an update of all filter lists.

> **Update now**  
> This button is available only if there is at least one filter list that is out of date. You can force an update of all outdated filter lists.
> 
> When a filter list gets updated using a newer version from its remote location, a clock icon will appear next to it. You can force an update of a single filter list by clicking the clock icon. Clicking this icon will reset the last updated timestamp, remove its content from storage, and cause the Update now button to become available.

> **Stock filter lists**
> This is a collection of various filter lists, grouped by purpose. To use a specific filter list, just select it through its checkbox. Any change in the selection of filter lists must be committed by using the Apply change button, which will appear if and only if the current selection of filter lists differs from the previous selection of filter lists.

**Reply by GitAcc53** [reply]
These doesnt answer any of my questions :(

**Reply by gothic-bum** [reply]
Your question was:  
  
> If purge updates filters, than what does update do?
  
That is specifically answered in the linked documentation and quotes that I provided. As are most (if not all) of your other questions.

**Reply by krystian3w** [reply]
As you don't use "thousands" of computers on a daily basis, you can click yourself once a day in clearing cache of the "EasyList Cookies"* or "AdGuard Cookies"** list to key in that the message has already been hidden/disabled correctly (so that the site doesn't break).

Unfortunately, there is no universal method to hide 100% of bars and pop-ups about cookies/GDPR/CCPA - if you detect a leak then just check these 3 cookie lists to see if they are updated to a newer version and if the bar/pop-up still doesn't disappear then consider reporting to both developers:

* https://github.com/easylist/easylist/issues - there can be delays of up to 2-6 months before someone checks what can be done with a pop-up or bar, a good practice would be to check if there is any open bug report about this site, then you can activate notifications for it.

* https://github.com/AdguardTeam/AdguardFilters#report-issues - It usually takes up to 2 weeks for someone to determine if the site is popular enough to get rid of the bar/pop-up on it, their dedicated form will dock your request to an open request if one exists, or do not artificially dock to a "www." link - as the site no longer uses it (such a URL is treated by the mechanism as a separate thread).


\* - certainly keep in mind that this list is updated on uBo's CDN every "4" hours or so despite the fact that EasyList provides an update every commit in the repository (Available for up to 10 minutes on their servers).

\** - certainly keep in mind that this list is updated on AdGuard server every 1 hour or so as there was a commit changing something.

---

As for the glowing message that it's possible to perform the update, it's one week after the release of the add-on anyway, and apparently something is further interfering with the automatic update to the point that it can't perform within 30 minutes of installing the add-on (this in turn may require good documentation of where the cause is).

---

As for the cookie message in such subdomains (consent.google.com / consent.youtube.com), this has not yet been addressed by anyone at uAssets.

> `https://consent.google.com/m?continue=https://news.google.com/&gl=UK&m=0&pc=n&cm=2&hl=en&src=1`

Fixed are only modal version without redirect.

**Reply by GitAcc53** [reply]
Thank you for the very detailed answer but I still didnt get any answers to any of my questions.

"It usually takes up to 2 weeks for someone to determine if the site is popular enough to get rid of the bar/pop-up on it"

All the websites I listed are popular websites like google, reddit, twitch. I dont understand this answer.

**Reply by krystian3w** [reply]
I meant AdGuard can reject address small pages or local web shops with very limited ship to one city/town by company "tir"/track car.

'new' Reddit and "twitch" use random classes in code so EL Cookies can add poor filters to list and skip update asap filter line stop working on both pages. Then the best check how addressed are both pages by AdGuard Cookies (maybe create more stable filters).

**Reply by GitAcc53** [reply]
Can you ELI5 how I am supposed to do that?

**Reply by krystian3w** [reply]
For new reddit  and twich check these:

```adb
twitch.tv##.consent-banner.consent-banner
reddit.com###SHORTCUT_FOCUSABLE_DIV > div > div > section > [style*="Toaster"][style*="Toaster"], shreddit-async-loader[bundlename*="cookie_banner"][bundlename*="cookie_banner"]
```

in my filters tab: https://github.com/gorhill/uBlock/wiki/Dashboard:-My-filters

How will it divide with 2 weeks then you can ask that EL cookies update the filters to the latest state of both domains.


The filters are based on IDCaC or Polish Cookies-GDPR that I use.

**Reply by GitAcc53** [reply]
Thats not ELI5 at all, I dont wanna mess around with the settings that much, I am not good with scripts. 

Is there another way other than uBlock filters to get rid of cookie notices? I thought about chrome extensions but "I don't care about cookies" doesnt work they say.

**Reply by gorhill** [reply]
> Thats not ELI5 at all

As per pattern observed so far,  I don't think you will ever be satisfied with whatever answers are given to you, and as a result you are burdening volunteers here. Consider that uBO might not be for you and you might be better using another content blocker. I don´t like to see volunteers being unduly burdened like this.
